### PR TITLE
Fix Environmental Metrics internal Error

### PR DESCRIPTION
**Pull request by lxwinspur**
Environmental Metrics has an internal error.

resolved by: https://github.com/ibm-openbmc/dev/issues/3401
upstream: https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/43170

Signed-off-by: Albert Zhang <zhanghaodi@inspur.com>
Change-Id: I6a75dfbc94f39d90a1368770076b874608394691

### CODE REVIEW

**Comment by lxwinspur** [comment]
@gtmills 
Please review this patch, thanks

**Comment by gtmills** [comment]
Looks reasonable. Thanks!!!!

File: redfish-core/lib/environment_metrics.hpp [hunk 1]
```diff
@@ -117,58 +115,100 @@ inline void
...
+                    judgment = true;
+                    break;
+                }
+            }
+            if (judgment)
+            {
+                const std::array<const char*, 1> totalPowerInterfaces = {
+                    "xyz.openbmc_project.Sensor.Value"};
+                const std::string& totalPowerPath =
+                    "/xyz/openbmc_project/sensors/power/total_power";
```
Comment (aahmed-2) [inline redfish-core/lib/environment_metrics.hpp:146]: We should get "totalPowerPath" from the above call instead of hardcoding it here.

Comment (aahmed-2) [reply redfish-core/lib/environment_metrics.hpp:146]: Something like:
insert at line 130:   const std::string& totalPowerPath = "";
...
insert at line 137: totalPowerPath = path;

File: redfish-core/lib/environment_metrics.hpp [hunk 2]
```diff
@@ -117,58 +115,100 @@ inline void
 inline void getPowerWatts(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
```
Comment (aahmed-2) [inline redfish-core/lib/environment_metrics.hpp:115]: I think we need to relook at the dbus calls here. I think we might be able to eliminate the outer dbus call and replace it with an error not found for total_power.

**Review by aahmed-2 — APPROVED** [review]
Did some basic curl get testing. Basic fields seemed to fill out correctly without internalError

**Review by aahmed-2 — APPROVED** [review]

**Review by gtmills — APPROVED** [review]
Defaulting some to Ali here. Seems reasonable though.

**Pull request merged**